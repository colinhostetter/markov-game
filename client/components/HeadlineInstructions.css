.headline-instructions-container {
  text-align: center;
  padding-bottom: 1em;
}
.headline-instructions-container h2 {
  margin: 0.5em;
}
.headline-instructions-container aside {
  margin: 0em;
}
