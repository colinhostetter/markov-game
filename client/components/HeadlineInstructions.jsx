import * as React from "react";
import "./HeadlineInstructions.css";

const HeadlineInstructions = ({ writing, guessing, round }) => {
  let headline, hint;
  if (round === 0) {
    headline = writing ? "You're writing first!" : "You're guessing first!";
  } else {
    headline = writing ? "Your turn to write!" : "Your turn to guess!";
  }
  hint = writing ? "Create a sentence your partner will recognize as human-written" :
                   "Guess which sentence is being written by your partner";
  return (
    <div className="headline-instructions-container padded-edges">
      <h2>{headline}</h2>
      <aside>{hint}</aside>
    </div>
  )
}

export default HeadlineInstructions;
