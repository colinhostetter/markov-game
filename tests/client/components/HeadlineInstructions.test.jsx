import React from 'react';
import HeadlineInstructions from '../../../client/components/HeadlineInstructions.jsx';
import renderer from 'react-test-renderer';

test("Round 1 writing", () => {
  const component = renderer.create(<HeadlineInstructions round={0} writing={true} guessing={false} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("Round 2+ writing", () => {
  const component = renderer.create(<HeadlineInstructions round={1} writing={true} guessing={false} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("Round 1 guessing", () => {
  const component = renderer.create(<HeadlineInstructions round={0} writing={false} guessing={true} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test("Round 2+ guessing", () => {
  const component = renderer.create(<HeadlineInstructions round={1} writing={false} guessing={true} />);
  expect(component.toJSON()).toMatchSnapshot();
});
