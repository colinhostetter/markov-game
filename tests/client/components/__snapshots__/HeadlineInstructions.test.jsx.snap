// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Round 1 guessing 1`] = `
<div
  className="headline-instructions-container padded-edges"
>
  <h2>
    You're guessing first!
  </h2>
  <aside>
    Guess which sentence is being written by your partner
  </aside>
</div>
`;

exports[`Round 1 writing 1`] = `
<div
  className="headline-instructions-container padded-edges"
>
  <h2>
    You're writing first!
  </h2>
  <aside>
    Create a sentence your partner will recognize as human-written
  </aside>
</div>
`;

exports[`Round 2+ guessing 1`] = `
<div
  className="headline-instructions-container padded-edges"
>
  <h2>
    Your turn to guess!
  </h2>
  <aside>
    Guess which sentence is being written by your partner
  </aside>
</div>
`;

exports[`Round 2+ writing 1`] = `
<div
  className="headline-instructions-container padded-edges"
>
  <h2>
    Your turn to write!
  </h2>
  <aside>
    Create a sentence your partner will recognize as human-written
  </aside>
</div>
`;
